sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/FilterPushdownSuite.scala

```diff
@@ -146,6 +146,32 @@ class FilterPushdownSuite extends PlanTest {
     comparePlans(optimized, correctAnswer)
   }
 
+  test("nondeterministic: can't push down filter through project") {
+    val originalQuery = testRelation
+      .select(Rand(10).as('rand))
+      .where('rand > 5)
```

Comment:
Maybe it is better to use a condition having both deterministic and non-deterministic expressions, e.g.
```
 val originalQuery = testRelation
  .select(Rand(10).as('rand), 'a)
  .where('rand > 5 && 'a > 5)
```